eth/helpers.go

```diff
@@ -121,3 +103,66 @@ func fromPerc(perc float64, multiplier *big.Float) *big.Int {
 	intRes, _ := floatRes.Int(nil)
 	return intRes
 }
+
+func ToBaseAmount(v string) (*big.Int, error) {
+	// check that string is a float represented as a string with "." as seperator
+	ok, err := regexp.MatchString("^[-+]?[0-9]*.?[0-9]+$", v)
+	if !ok || err != nil {
+		return nil, fmt.Errorf("submitted value %v is not a valid float", v)
+	}
+	splitNum := strings.Split(v, ".")
+	// If '.' is absent from string, add an empty string to become the decimal part
+	if len(splitNum) == 1 {
+		splitNum = append(splitNum, "")
+	}
+	intPart := splitNum[0]
+	decPart := splitNum[1]
+
+	// If decimal part is longer than 18 decimals, trim the string and round off
+	if len(decPart) > 18 {
+		last := int(decPart[19])
```

Comment:
A value with greater than 18 decimal places would be valid so I think we can just return an error. Should add a test that checks for this.